diff --git a/lib/apache b/lib/apache
index 41c2e3d..ab5e668 100644
--- a/lib/apache
+++ b/lib/apache
@@ -123,8 +123,11 @@ function install_apache_uwsgi {
         pip_install uwsgi
         pip download uwsgi -c $REQUIREMENTS_DIR/upper-constraints.txt
         local uwsgi
-        uwsgi=$(ls uwsgi*)
-        tar xvf $uwsgi
+        #uwsgi=$(ls uwsgi*)
+        #tar xvf $uwsgi
+        if [ -f uW*.tar.gz ] ; then mv -f uW*.tar.gz uwsgi-latest.tar.gz; mkdir uwsgi-latest ; fi
+        uwsgi=$(ls uwsgi*.tar.gz)
+        tar xvf $uwsgi -C uwsgi-latest
         cd uwsgi*/apache2
         sudo $apxs -i -c mod_proxy_uwsgi.c
         popd
